src/dmd/expressionsem.d

```diff
@@ -11602,6 +11602,233 @@ Expression binSemanticProp(BinExp e, Scope* sc)
     e.e2 = e2x;
     return null;
 }
+/********************************************************************************
+* Helper function to resolve `@property` functions in a `BinAssignExp`.
+* It rewrites expressions of the form `e1.prop @= e2` to `e1.x(e1.x @ e2)`
+* if 'e1` is a type, otherwise it rewrites to
+* `((auto ref _e1) => _e1.prop(_e1.prop @ e2))(e1)`
+* Params:
+*      e = the binary assignment expression to rewrite
+*      sc = the semantic scope
+* Returns:
+*      the rewritten expression if the procedure succeeds, an `ErrorExp` if the
+*      and error is encountered, or `null` if `e.e1` is not a `@property` function.
+*/
+Expression SemanticProp(BinAssignExp e, Scope* sc)
```

Comment:
Isn't this called `binSemanticProp`?

Reply:
It used to be called that. It was initially to be implemented for binary operators only, which I plan to expand father than that. That may change in future.